avm/ptn/app/container-job-toolkit/modules/deploy_services.bicep

```
@@ -0,0 +1,930 @@
+    infrastructureResourceGroupName: '${resourceGroupName}-infrastructure'
+    // vnet configuration
+    internal: deployInVnet ? true : false
+    infrastructureSubnetId: deployInVnet || !empty(addressPrefix) ? vnet.outputs.subnetResourceIds[2] : null // third subnet is the workload subnet
+    dockerBridgeCidr: deployInVnet ? dockerBridgeCidr : null
+    platformReservedCidr: deployInVnet ? platformReservedCidr : null
+    platformReservedDnsIP: deployInVnet ? platformReservedDnsIP : null
+  }
+}
+
+output logAnalyticsResourceId string = !empty(logAnalyticsWorkspaceResourceId)
+  ? logAnalyticsWorkspaceResourceId!
+  : law.id
+output vaultName string = vault.outputs.name
+output keyVaultAppInsightsConnectionStringUrl string = !empty(appInsightsConnectionString ?? '')
+  ? '${vault.outputs.uri}/secrets/applicationinsights-connection-string'
+  : ''
+output registryName string = registry.outputs.name
+output registryLoginServer string = registry.outputs.loginServer
+output managedEnvironmentId string = managedEnvironment.outputs.resourceId
+output userManagedIdentityResourceId string = managedIdentityResourceId == null
+  ? userIdentity_new.id
+  : userIdentity_existing.id
+output userManagedIdentityPrincipalId string = managedIdentityResourceId == null
+  ? userIdentity_new.properties.principalId
+  : userIdentity_existing.properties.principalId
+output storageAccountResourceId string = deployInVnet ? storage.outputs.resourceId : ''
+
+output vnetResourceId string = deployInVnet ? vnet.outputs.resourceId : ''
+
+output privateEndpointSubnetAddressPrefix string = deployInVnet ? privateEndpointSubnetAddressPrefix : ''
+output deploymentscriptSubnetAddressPrefix string = deployInVnet ? deploymentscriptSubnetAddressPrefix : ''
+output workloadSubnetAddressPrefix string = deployInVnet ? workloadSubnetAddressPrefix : '' // also used for zoneRedundant configuration
+output subnetResourceId_deploymentScript string = deployInVnet ? vnet.outputs.subnetResourceIds[1] : ''
+
+output dockerBridgeCidr string = dockerBridgeCidr
+output platformReservedCidr string = platformReservedCidr
+output platformReservedDnsIP string = platformReservedDnsIP
+
+// ================ //
+// Definitions      //
+// ================ //
+
+@export()
+type secretType = {
+  @description('Optional. Resource ID of a managed identity to authenticate with Azure Key Vault, or System to use a system-assigned identity.')
+  identity: string?
+
+  @description('Conditional. Azure Key Vault URL pointing to the secret referenced by the Container App Job. Required if `value` is null.')
+  @metadata({
+    example: '''https://myvault${environment().suffixes.keyvaultDns}/secrets/mysecret'''
+  })
+  keyVaultUrl: string?
+
+  @description('Optional. The name of the secret.')
+  name: string?
+
+  @description('Conditional. The secret value, if not fetched from Key Vault. Required if `keyVaultUrl` is not null.')
+  @secure()
+  value: string?
+}
+
+@export()
+type securityRulesType = {
```

Please keep UDT singular and not nullable
```suggestion
type securityRuleType = {
```